compiler/luci/pass/src/helpers/ArrayIndex.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2025 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __LUCI_ARRAY_INDEX_H__
+#define __LUCI_ARRAY_INDEX_H__
+
+#include <cstdint>
+
+namespace luci
+{
+
+/**
+ * @brief Index class for 4D tensor (NHWC) to calculate linear index from multi-dimensional indices.
```

Comment:
`NHWC`, below functions use `NHWD`.
it would be better to use one convention not to get confused.

Reply:
I used d0, d1, .. convention in cpp. Let me change it into the same dX convention not to cause confusion as you said. + integrity